rskj-core/src/main/java/co/rsk/core/bc/TransactionConflictDetector.java

```diff
@@ -0,0 +1,367 @@
+
+        checkAccountAccessed(account, partitionId);
+        if (!checkAccountDeleted(account, partitionId)) {
+            trackAccountDeleted(account, partitionId);
+        }
+    }
+
+    private Conflict createConflict(ByteArrayWrapper key, int conflictPartId, int originPartId, eConflictType conflictType) {
+        TransactionsPartition conflictPartition = partitioner.fromId(conflictPartId);
+        TransactionsPartition originPartition = partitioner.fromId(originPartId);
+        if (conflictPartition == null) {
+            logger.error("Unable to get the partition for id [{}]", conflictPartId);
+        }
+        if (originPartition == null) {
+            logger.error("Unable to get the partition for id [{}]", originPartId);
+        }
+        return new Conflict(key, conflictPartition, originPartition, conflictType);
+    }
+
+    protected void recordConflict(Conflict conflict) {
+        Collection<Conflict> conflicts;
+        conflicts = conflictsPerPartition.computeIfAbsent(conflict.getConflictFrom(), k -> new HashSet<>());
+        conflicts.add(conflict);
+        conflicts = conflictsPerPartition.computeIfAbsent(conflict.getConflictWith(), k -> new HashSet<>());
+        conflicts.add(conflict);
+    }
+
+    private boolean checkKeyAlreadyWritten(ByteArrayWrapper key, int partId) {
+        Integer otherPartId = partitionWriterPerKey.get(key);
+        if ((otherPartId == null) || otherPartId.equals(partId)) {
+            return false;
+        } else {
+            logger.info("key [{}] has already been written by partition [{}] --> conflict from partition [{}]",
+                    key, otherPartId, partId);
+            recordConflict(createConflict(key, otherPartId, partId, eConflictType.WRITTEN_BEFORE_ACCESSED));
+        }
+        return true;
+    }
+
+    private boolean checkKeyAlreadyRead(ByteArrayWrapper key, int partitionId) {
+        Collection<Integer> otherPartIds = partitionReadersPerKey.get(key);
+        if (otherPartIds == null) {
+            return false;
+        }
+        // there can not be a conflict with itself
+        otherPartIds.remove(partitionId);
+        if (otherPartIds.isEmpty()) {
+            return false;
+        }
+        // Record a conflict for each reading group
+        otherPartIds.forEach(conflictPartId -> {
+            recordConflict(createConflict(key, conflictPartId, partitionId, eConflictType.ACCESSED_BEFORE_WRITTEN));
+            logger.info("key [{}] has already been read by partition [{}] --> conflict from partition [{}]",
+                    key, conflictPartId, partitionId);
+        });
+        return true;
+    }
+
+    private boolean checkAccountAccessed(ByteArrayWrapper account, int partitionId) {
+        Collection<Integer> otherPartIds = accessedAccounts.get(account);
+        if (otherPartIds == null) {
+            return false;
+        }
+        // there can not be a conflict with itself
+        otherPartIds.remove(partitionId);
```

Comment:
side effects here too

Reply:
Here again, the only case when you remove the current partition from the accessors list of the account is when the same partition is deleting this account.
So, in the end there will be a conflict with any other partition that would attempt to access the same account, even if the accessors list has been purged (unintentionally I confess).
Anyway, I fix the bug here too, and I take this opportunity to factorize this code to reduce duplication.